src/Raven.Server/Documents/Indexes/Static/IndexedDocumentsEnumerator.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace Raven.Server.Documents.Indexes.Static
+{
+    public class IndexedDocumentsEnumerator : IEnumerable<DynamicDocumentObject>
+    {
+        private readonly IEnumerable<Document> _docs;
+        private readonly DynamicDocumentObject _dynamicDocument = new DynamicDocumentObject();
+
+        public IndexedDocumentsEnumerator(IEnumerable<Document> docs)
+        {
+            _docs = docs;
+        }
+
+        public IEnumerator<DynamicDocumentObject> GetEnumerator()
```

Comment:
You mean to create own implementation of `IEnumerator` and handle `MoveNext` method, don't you?